src/Http/Http.Extensions/gen/RequestDelegateGenerator.cs

```diff
@@ -146,11 +146,61 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                 return stringWriter.ToString();
             });
 
-        var thunksAndEndpoints = thunks.Collect().Combine(stronglyTypedEndpointDefinitions);
+        var endpointHelpers = endpoints
+            .Collect()
+            .Select((endpoints, _) =>
+            {
+                var parameters = endpoints.SelectMany(endpoint => endpoint.Parameters);
+                var responses = endpoints.Select(endpoint => endpoint.Response);
+
+                var hasJsonBodyOrService = parameters.Any(parameter => parameter.Source == EndpointParameterSource.JsonBodyOrService);
+                var hasJsonBody = parameters.Any(parameter => parameter.Source == EndpointParameterSource.JsonBody);
+                var hasRouteOrQuery = parameters.Any(parameter => parameter.Source == EndpointParameterSource.RouteOrQuery);
+                var hasBindAsync = parameters.Any(parameter => parameter.Source == EndpointParameterSource.BindAsync);
+                var hasParsable = parameters.Any(parameter => parameter.IsParsable);
+                var hasJsonResponse = responses.Any(response => !response.ResponseType.IsSealed && !response.ResponseType.IsValueType);
+
+                using var stringWriter = new StringWriter(CultureInfo.InvariantCulture);
+                using var codeWriter = new CodeWriter(stringWriter, baseIndent: 0);
+
+                if (hasRouteOrQuery)
```

Comment:
I guess I would have thought to do this differently.

Instead of post-processing all the endpoints. As we are emitting the other code, I would have thought we would put a boolean flag for each "optional" thing we would need. For example, when we are emitting a call to `TryParseExplicit`, we would set the boolean that said "we now require TryParseExplicit code to be generated".

Then at the end, inspect all the booleans that got set during the processing, and generate each helper.